packages/providers/src/provider.ts

```diff
@@ -341,6 +341,9 @@ export default class Provider {
    *
    * If there are missing VariableOutputs
    * `addVariableOutputs` is called on the transaction.
+   *
+   * If there are missing output contract IDs, those are fetched and added to the transaction.
+   *
```

Comment:
Is this functionality working? When I was rewriting the `Inter-contract Calls` doc page I was always running into an error before specifically calling the `addContracts` function in the contract function chain call.

Reply:
@Torres-ssf there seems to be code in place that would take care of that functionality, but we don't have any tests in place for this. 

https://github.com/FuelLabs/fuels-ts/blob/dadc5760549564480b7b580cb5c268030bf7dcd6/packages/providers/src/provider.ts#L374-L376

Reply:
@Torres-ssf thanks for looking into this with so much detail. Unfortunately, I don't have a lot of insight on the dry-run implementation, but this is def something we need to fix. I can file a follow-up issue for this. Does that sound good?

Reply:
@Dhaiwat10 Yes that sounds great